src/main/kotlin/org/elm/lang/core/resolve/scope/ModuleScope.kt

```diff
@@ -47,27 +63,32 @@ class ModuleScope(val elmFile: ElmFile) {
 
     // VALUES
 
-
     fun getDeclaredValues(): List<ElmNamedElement> {
-        val valueDecls = elmFile.getValueDeclarations().flatMap {
-            it.declaredNames(includeParameters = false)
+        return CachedValuesManager.getCachedValue(elmFile, DECLARED_VALUES_KEY) {
+            val valueDecls = elmFile.getValueDeclarations().flatMap {
+                it.declaredNames(includeParameters = false)
+            }
+            val values = listOf(valueDecls, elmFile.getPortAnnotations(), elmFile.getInfixDeclarations())
+                    .flatten()
+            Result.create(values, elmFile.project.modificationTracker)
```

Comment:
If you change a declaration, then anything that references that declaration needs to be invalidated to avoid e.g. incorrect inference results. Since elements outside of this file can reference it, we need to invalidate everything.